src/app/organization/employees/create-employee/create-employee.component.ts

```diff
@@ -0,0 +1,91 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+
+/** Custom Services */
+import { OrganizationService } from '../../organization.service';
+
+/**
+ * Create employee component.
+ */
+@Component({
+  selector: 'mifosx-create-employee',
+  templateUrl: './create-employee.component.html',
+  styleUrls: ['./create-employee.component.scss']
+})
+export class CreateEmployeeComponent implements OnInit {
+
+  /** Minimum joining date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum joining date allowed. */
+  maxDate = new Date();
+  /** Employee form. */
+  employeeForm: FormGroup;
+  /** Office data. */
+  officeData: any;
+
+  /**
+   * Retrieves the offices data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {OrganizationService} organizationService Organization Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private organizationService: OrganizationService,
+              private route: ActivatedRoute,
+              private router: Router) {
+    this.route.data.subscribe((data: { offices: any }) => {
+      this.officeData = data.offices;
+    });
+  }
+
+  /**
+   * Creates the employee form.
+   */
+  ngOnInit() {
+    this.createEmployeeForm();
+  }
+
+  /**
+   * Creates the employee form.
+   */
+  createEmployeeForm() {
+    this.employeeForm = this.formBuilder.group({
+      'officeId': ['', Validators.required],
+      'firstname': ['', Validators.required],
+      'lastname': ['', Validators.required],
```

Comment:
Refer to the community app's code. Another validation is supposed to be put here. Update accordingly.

Reply:
Added the implemented validator from the community app (First letter must be a A-z, the second letter can be anything). 

The regex in the community app is different, but it seems like angularJS treats regex slightly differently than here. The regex in the community app actually matches every character for the second letter, but the code is supposed to match A-z, . , - and '. 

When I applied the same regex to here it invalidated all special characters apart from .,- and numbers, so I implemented this instead to match the actual behaviour in the community app.